Public/Sdk/Experimental/NodeJs/npm.dsc

```diff
@@ -41,14 +41,43 @@ namespace Npm {
         };
     }
 
+    @@public
+    export function npmInstall(workingDirectory: StaticDirectory): OpaqueDirectory {
+        const wd = workingDirectory.root;
+        const nodeModulesPath = d`${wd}/node_modules`;
+        const npmCachePath = Context.getNewOutputDirectory('npm-install-cache');
+
+        const arguments: Argument[] = [
+            Cmd.argument(Artifact.input(Node.npmCli)),
+            Cmd.argument("install"),
+            Cmd.option("--cache ", Artifact.none(npmCachePath)), // Forces the npm cache to use this output folder for this object so that it doesn't write to user folder
+        ];
+
+        const result = Node.run({
+            arguments: arguments,
+            workingDirectory: wd,
+            dependencies: [ workingDirectory ].filter(x => x !== undefined),

```

Comment:
>workingDirectory [](start = 28, length = 16)

in what case this will be undefined?

Reply:
it's never undefined and should be changed to 
```suggestion
            dependencies: [ workingDirectory ],
```